src/event_sources/ble_esp32.h

```diff
@@ -102,6 +106,92 @@ class GATTResource : public BLEResource {
   struct os_mbuf* _mbuf = null;
 };
 
+
+class BLEServerServiceResource;
+typedef LinkedList<BLEServerServiceResource> BLEServerServiceList;
+
+class BLEServerCharacteristicResource;
+typedef LinkedList<BLEServerCharacteristicResource> BLEServerCharacteristicList;
+
+class BLEServerCharacteristicResource: public Resource, public BLEServerCharacteristicList::Element{
+public:
+  TAG(BLEServerCharacteristicResource);
+  explicit BLEServerCharacteristicResource(ResourceGroup *resource_group, BLEServerServiceResource* service,
+                                           ble_uuid_any_t uuid, int type, struct os_mbuf *value):
+      Resource(resource_group),
+      _service(service),
+      _uuid(uuid),
+      _type(type),
+      _mbuf_to_send(value),
+      _nimble_value_handle(0),
+      _mbuf_received(null),
+      _indicate(false),
+      _notify(false),
+      _conn_handle(0),
+      _mutex(OS::allocate_mutex(3, "")) {}
+
+  ble_uuid_any_t uuid() const { return _uuid; }
+  ble_uuid_t *uuid_p() { return &_uuid.u; }
+  int type() const { return _type; }
+  struct os_mbuf *mbuf_to_send() { Locker locker(_mutex); return _mbuf_to_send; };
+  void set_mbuf_to_send(struct os_mbuf* mbuf);
+  uint16_t *ptr_nimble_value_handle() { return &_nimble_value_handle; }
+  uint16_t nimble_value_handle() const { return _nimble_value_handle; }
+  bool is_notify_enabled() const { return _notify; }
+  bool is_indicate_enabled() const { return _indicate; }
+  uint16 conn_handle() const { return _conn_handle; }
+  void set_mbuf_received(os_mbuf *mbuf);
+  os_mbuf *mbuf_received() { Locker locker(_mutex); return _mbuf_received; }
+
+  void set_subscription_status(bool indicate, bool notify, uint16 conn_handle) {
+    _indicate = indicate;
+    _notify = notify;
+    _conn_handle = conn_handle;
+  };
+
+private:
+  BLEServerServiceResource* _service;
+  ble_uuid_any_t _uuid;
+  int _type;
+  struct os_mbuf *_mbuf_to_send;
+  uint16 _nimble_value_handle;
+  os_mbuf *_mbuf_received;
+  bool _indicate;
+  bool _notify;
+  uint16 _conn_handle;
+  Mutex *_mutex;
```

Comment:
`Mutex* _mutex` (this is general and there are more cases)

Reply:
I have tries to change all occurences of type *var to type* var and <type *> to <type*>. To make it consistant across the whole cc file, i also changed some of the existing occurences